pkg/vsphere/disk/disk_manager.go

```diff
@@ -281,7 +288,9 @@ func (m *Manager) Attach(op trace.Operation, disk *types.VirtualDisk) error {
 	machineSpec := types.VirtualMachineConfigSpec{}
 	machineSpec.DeviceChange = append(machineSpec.DeviceChange, changeSpec...)
 
-	m.reconfig.Lock()
+	m.mu.Lock()
+	defer m.mu.Unlock()
```

Comment:
yeah - I'd mentioned that in an as yet unsubmitted commit. Given it's now against an outdated diff:
> A single idempotent entry point for having the disk attached/detached that reference counts and "does the right thing" seems less convoluted. Is there a reason you've found for calling setAttached separately from the Attach call? I've not run through the code explicitly looking for a reason so forgive me if it's something obvious.